graphix/fundamentals.py

```diff
@@ -0,0 +1,307 @@
+"""Fundamental components related to quantum mechanics."""
```

Comment:
is it correct to understand the motivation for creating this file is to avoid circular import?

Reply:
Conceptually all this can sit in `pauli.py` but if that cause circular imports, then it makes sense. in such a case, different module name like `basic_algebra` might be good, `fundamental` sound a bit too profound?

Reply:
>  to avoid circular import

Exactly!

> `basic_algebra`

I feel it's a little bit odd since this module is full of `Enum`s. Is `Enum` related to `algebra`?

Reply:
`pauli_base`? I'm stil wondering why we had to split into fundamentals.py and pauli.py, btw. where did you see the circular import?

Reply:
`pauli_base` looks plausible to me, while some features could be rather irrelevant to Pauli...


> where did you see the circular import?

Actually we can still break most of import loops without this modification, but I dare to change it so that basic components can be used everywhere possible without import circulation.
For more concrete example, please check how `_db.CLIFFORD_MEASURE` was updated and also [this thread](https://github.com/TeamGraphix/graphix/pull/216#pullrequestreview-2347999429) .